pcsx2/Sio.cpp

```diff
@@ -122,9 +121,12 @@ void sioInit()
 bool isR3000ATest = false;
 
 // Check the active game's type, and fire the matching interrupt.
+// The HW_IFCG register lets us know if PSX mode is active.
+// The "3rd bit" is supposed to be the flag, but the actual value evaluates to 8 for PSX.
+// Conversely, the value is 1 for PS2. 
 // Note that the R3000A's call to interrupts only calls the PS2 based (lack of) delays.
 SIO_FORCEINLINE void sioInterrupt() {
-	if (psxMode && !isR3000ATest) {
+	if (psxHu32(HW_ICFG) == 8 && !isR3000ATest) {
```

Comment:
// check if bit 3 is set
if ((psxHu32(HW_ICFG) & (1<<3)) && !isR3000ATest) {

Reply:
This way we don't care about the other bits, so the register can be something other than evaluating exactly to 8.
Nice that you found it though! :)